client/components/TestQueueRun/index.jsx

```diff
@@ -8,217 +9,275 @@ import {
+                                    ? 'assigned'
+                                    : 'not-assigned';
+                                return (
+                                    <Dropdown.Item
+                                        variant="secondary"
+                                        as="button"
+                                        key={nextId()}
+                                        onClick={async () => {
+                                            await toggleTesterAssign(
+                                                tester.username
+                                            );
+                                            setAlertMessage(
+                                                `You have been ${
+                                                    classname.includes('not')
+                                                        ? 'removed from'
+                                                        : 'assigned to'
+                                                } this test run.`
+                                            );
+                                        }}
+                                        aria-checked={isTesterAssigned}
+                                        role="menuitemcheckbox"
+                                    >
+                                        {isTesterAssigned && (
+                                            <FontAwesomeIcon icon={faCheck} />
+                                        )}
+                                        <span className={classname}>
+                                            {`${tester.username}`}
+                                        </span>
+                                    </Dropdown.Item>
+                                );
+                            })
+                        ) : (
+                            <span className="not-assigned">
+                                No testers to assign
+                            </span>
+                        )}
+                    </Dropdown.Menu>
+                </Dropdown>
+            </>
         );
-    }
-
-    renderTesterList(currentUserAssigned) {
-        const { testers, userId } = this.props;
-
-        let userInfo = testers
-            .filter(uid => uid !== userId)
-            .map(uid => this.renderTestsCompletedByUser(uid));
-        if (currentUserAssigned) {
-            userInfo.unshift(this.renderTestsCompletedByUser(userId));
-        }
+    };
 
-        return userInfo;
-    }
+    const evaluateTestRunTitle = () => {
+        const { testPlanTarget, testPlanVersion } = testPlanReport;
+        const { title: testPlanTargetName } = testPlanTarget;
+        const { title: apgExampleName, directory } = testPlanVersion;
 
-    renderOpenAsDropdown() {
-        const { runId, testers, usersById } = this.props;
+        return `${apgExampleName || directory} for ${testPlanTargetName}`;
+    };
```

Comment:
This seems like business logic so maybe we can add it to GraphQL - actually I thought maybe the database had this field already? Anyway I think this probably makes more sense on the server, or else we might end up repeating this function in multiple places.

Reply:
It has for the constructed TestPlanTarget title - `<at> <atVersion> with <browser> <browserVersion>`, which is `testPlanTargetName` is. There isn't a combination for it with the Test Plan's title, or in the absence of the Test Plan's title, the actual test plan (directory) coming from the server though. We could definitely do that that if we don't want to create a utility method within the frontend for it